gpii/node_modules/gpii-oauth2/gpii-oauth2-datastore/src/DbDataStore.js

```diff
@@ -0,0 +1,614 @@
+        },
+        findAuthDecisionsByGpiiToken: {
+            funcName: "gpii.oauth2.dbDataStore.findRecord",
+            args: [
+                "{that}.findAuthDecisionsByGpiiTokenDataSource",
+                {
+                    gpiiToken: "{arguments}.0"
+                },
+                "gpiiToken",
+                gpii.oauth2.dbDataStore.handleMultipleRecords
+            ]
+            // gpiiToken
+        },
+        findAuthDecision: {
+            funcName: "gpii.oauth2.dbDataStore.findRecord",
+            args: [
+                "{that}.findAuthDecisionDataSource",
+                {
+                    gpiiToken: "{arguments}.0",
+                    clientId: "{arguments}.1",
+                    redirectUri: "{arguments}.2"
+                },
+                ["gpiiToken", "clientId", "redirectUri"]
+            ]
+            // gpiiToken, clientId, redirectUri
+        },
+        // TODO make authCodes active only for a limited period of time
+        // TODO make authCodes single use
+        saveAuthCode: {
+            funcName: "gpii.oauth2.dbDataStore.saveAuthCode",
+            args: [
+                "{that}.saveDataSource",
+                "{arguments}.0",
+                "{arguments}.1"
+            ]
+            // authDecisionId, code
+        },
+        findAuthByCode: {
+            funcName: "gpii.oauth2.dbDataStore.findRecord",
+            args: [
+                "{that}.findAuthByCodeDataSource",
+                {
+                    code: "{arguments}.0"
+                },
+                "code",
+                gpii.oauth2.dbDataStore.findAuthByCodePostProcess
+            ]
+            // code
+        },
+        // Note: With the In Memory Data Store, this function returns an empty array when no clients are found.
+        // However, with the DB Data Store, "undefined" is returned.
+        findAuthorizedClientsByGpiiToken: {
+            funcName: "gpii.oauth2.dbDataStore.findRecord",
+            args: [
+                "{that}.findAuthorizedClientsByGpiiTokenDataSource",
+                {
+                    gpiiToken: "{arguments}.0"
+                },
+                "gpiiToken",
+                gpii.oauth2.dbDataStore.findAuthorizedClientsByGpiiTokenPostProcess
+            ]
+            // gpiiToken
+        },
+        findAuthByAccessToken: {
+            funcName: "gpii.oauth2.dbDataStore.findRecord",
+            args: [
+                "{that}.findAuthByAccessTokenDataSource",
+                {
+                    accessToken: "{arguments}.0"
+                },
+                "accessToken",
+                gpii.oauth2.dbDataStore.findAuthByAccessTokenPostProcess
+            ]
+            // accessToken
+        },
+        // TODO: The concept of allowDirectGpiiTokenAccess flag was added to integrate Mobile Accessibility
+        // for the January 2015 review. It needs to be reassessed.
+        findAccessTokenByOAuth2ClientIdAndGpiiToken: {
```

Comment:
@amb26, after having a closer look at this function, I realized I was mistaken to think this function handles more than the case with `allowDirectGpiiTokenAccess` flag. After double checking with @simonbates, I've remove this function along with the use of `allowDirectGpiiTokenAccess` flag and also updated all related files.